### DISCUSSION TITLE
Best practices for using wiki links?

### DISCUSSION THREAD

**Discussion by Masstronaut** [pngwn/MDsveX#316] [Q&A]
My markdown content authoring workflow makes extensive use of `[[wiki links]]` which I would like to have as hyperlinks in the final rendered webpage. As I understand it, this requires:
- A list of all valid titles/aliases and their corresponding links/slugs
- The relationships between those links and titles so the markdown processor knows to link `[[My super awesome post]]` to `/posts/super_awesome`
- A remark or rehype plugin that transforms the parsed wikilinks into HTML hyperlinks that go to the correct page with the correct link text, ie `<a href="/posts/super_awesome">My super awesome post</a>`.

I've look at [remark-wiki-link](https://github.com/landakram/remark-wiki-link), but I'm not sure how to collect the list of page slugs and titles. It lives in the frontmatter of my markdown content, which means the content needs to be processed before I can process it.

Is there a way to glob data from the frontmatter of *every* markdown file and pass it to the remark-wiki-link plugin?

**Comment by jamesb93** [comment]
I'd be interested in this as well. We're trying to build a learning platform for some electronic musicians toolkits and being able to inter-reference glossary terms is on our todo list.

**Reply by Masstronaut** [reply]
Not an answer yet per se, but [this thread](https://github.com/pngwn/MDsveX/discussions/354#discussioncomment-1509591) may have some info that makes this a little easier. I still haven't gotten to it yet though.
